### PR TITLE
Enable ci-integration tests on GHA (incl fixing a few tests)

### PR DESCRIPTION
**Pull request by cognifloyd**
Move integration tests to GitHub Actions (GHA)

We also have to fix a few tests to make the transition.

List of commits:

- improve CI output
- improve GHA folding
- fix virtualenv lock files issue again
- Switch tests to check w/ CA that is present in GHA
- apt download cache
- use bitnami/rabbitmq service container in GHA
- bust python cache
- Use virtualenv from st2 virtualenv by default
- Installing virtualenv in --user on github
- Add workaround for failing stdin test
- fix check-dependency-conflicts by updating pip-tools

### CODE REVIEW

**Comment by cognifloyd** [comment]
Historical note:
I built this in st2sandbox/st2#1 then squashed/reordered the 100+ commits into a few discreet commits in st2sandbox/st2#2. In squashing/reordering commits, I removed most of the history of places I added various forms of debug logging.

Once this is merged, a maintainer should disable StackStorm/st2 on travis-ci.com. A future PR can clean up the repo, removing travis config, and getting rid of anything else that is travis-specific.

**Comment by cognifloyd** [comment]
failure in integration tests is one of the orquesta tests that have been flaky lately. Please restart GitHub Actions workflows.

**Comment by cognifloyd** [comment]
Fixes a bug with the cache. All tests pass now.

File: .github/workflows/ci.yaml
Comment (Kami) [inline .github/workflows/ci.yaml:211]: In the future I think it would be better to move that code in ``scripts/ci/setup-rabbitmq.sh`` or similar so we can also run shell check, etc on those files.

File: tools/launchdev.sh
Comment (Kami) [inline tools/launchdev.sh:159]: I guess we could even use ``--depth 1`` // swallow clone to speed things up, but then again that repo is not that bug.

File: scripts/travis/prepare-integration.sh
Comment (Kami) [inline scripts/travis/prepare-integration.sh:20]: As per comment above since that script is now specific to gha, we should probably rename the directory and update the affected files :)

**Comment by Kami** [comment]
Thanks for doing this, it's great to how all the tests now run on GHA.

As far as orquesta test failures go - yeah, it appears that some orquesta integration tests which were failing quite often on Travis are also failing quite often on GHA.

I already assumed before it's some kind of race condition and now it looks even more likely.

@m4dcoder Do you think setting up a coordination backend for integration tests would make a difference with those race conditions / occasional failures? Or it's likely some other root cause we should look at improving in tests / similar?

Here are some examples:

* https://travis-ci.com/github/StackStorm/st2/builds/220669792
* https://travis-ci.com/github/StackStorm/st2/builds/220020111
* https://travis-ci.com/github/StackStorm/st2/builds/220015532 - here they also hand and timeout, perhaps some deadlock?

**Comment by armab** [comment]
Wohoo! Congrats @cognifloyd, great achievement here that helps everyone a lot! :metal:

**Review by armab — APPROVED** [review]
:heavy_check_mark: integration tests for the GH Actions is the best indicator for me :+1: 
Great work!

Quick notes:
`ci.yaml` will need a bit of cleanup from the irrelevant comments and notes. Can we also remove Travis.yml?

**Comment by cognifloyd** [comment]
Agreed, we need to do quite a bit of cleanup to get rid of all the remnants of travis.

The scope of this PR is to do the bare minimum to get it working in GHA. I do not want to disable Travis in this PR to demonstrate that it works in both CI systems. Once merged, a Maintainer can go into the dashboard on travis-ci.com to disable running Travis on this repo. 

Also, I tried to document in `ci.yaml` why I made various decisions so that when we do cleanup that file, we don't inadvertently break things by going back to something that didn't work. That said, there is a lot of room for improvement with the ci files. I definitely want to see better organization with some scripts mixing out of the workflow definition and others just going away as we make them obsolete.

Please, let's save a lot of this cleanup for future PRs.

File: scripts/travis/prepare-integration.sh
Comment (cognifloyd) [reply scripts/travis/prepare-integration.sh:20]: It's not very difficult to make this work across multiple CI providers (like I did in https://github.com/ansible-community/molecule/pull/2976 ), but I didn't see the point since this will only be running in GHA for now.

As I don't want to disable Travis until after this PR is merged, I minimized my changes to the travis scripts. Once we disable travis, though, there's a lot of code on the chopping block.

File: contrib/runners/python_runner/python_runner/python_action_wrapper.py
Comment (blag) [inline contrib/runners/python_runner/python_runner/python_action_wrapper.py:366]: Not a blocker, but noting this in the future: it would be a good idea to log what already has been read into `stdin_data`.

```suggestion
            raise ValueError(f"Received no valid parameters data from sys.stdin:\n{stdin_data}")
```

**Review by blag — APPROVED** [review]

**Pull request merged**
